openmedia/gui/frames.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+
+import Tkinter as Tk
+import os, widgets, tkFileDialog
+from openmedia.player import mixer
+
+class InvalidWidgetStateException(Exception):
+    pass
+
+class PlayerFrame(Tk.Frame):
+    def __init__(self, root):
+        Tk.Frame.__init__(self, root)
+        self.f_control = _ControlFrame(self)
+        self.f_progress = _ProgressFrame(self)
+        self.f_playlist = _PlaylistFrame(self)
+        self.f_control.grid(row=0, column=0)
+        self.f_progress.grid(row=1, column=0)
+        self.f_playlist.grid(row=0, column=1, rowspan=10)
+        for i in xrange(0, 2):
+            self.grid_columnconfigure(i, weight=1)
+            self.grid_rowconfigure(i, weight=1)
+        self._bind_all()
+        mixer.play()
+
+    def _bind_all(self):
+        for button in self.f_control.buttons:
+            button.bind(button.sequence, getattr(self, '_' + button.name + '_handler'))
+        for button in self.f_playlist.buttons:
+            button.bind(button.sequence, getattr(self, '_' + button.name + '_handler'))
+        self.f_control.s_volume.bind('<B1-Motion>', self._set_volume)
+        self.f_progress.s_progress.bind('<ButtonRelease-1>', self._skip)
+        self.f_playlist.l_playlist.bind('<Double-Button-1>', self._play)
+        self.f_progress.set_alarm(self._progress)
+
+    def _update_play_icon(self, button):
+        if mixer.is_playing():
+            button.config(text=u'▶')
+        else:
+            button.config(text=u'▌▌')
+
+    def _play_pause(self, button):
+        if mixer.is_playing():
```

Comment:
We could call `_update_play_icon` here and anywhere else we need the text of the play button to change. It's up to you, I'm not even sure we need `_update_play_icon` anymore. Any thoughts?